icelake/src/io/appender/base_file_writer.rs

```diff
@@ -0,0 +1,377 @@
+//! A module provide `RollingWriter`.
+use arrow_array::RecordBatch;
+use arrow_cast::cast;
+use async_trait::async_trait;
+use std::sync::Arc;
+
+use crate::config::RollingWriterConfig;
+use crate::io::location_generator::FileLocationGenerator;
+use crate::io::{
+    FileWriteResult, FileWriter, FileWriterBuilder, IcebergWriteResult, IcebergWriter,
+    IcebergWriterBuilder, SingletonWriter,
+};
+
+use crate::{Error, Result};
+use arrow_schema::{DataType, SchemaRef};
+
+#[derive(Clone)]
+pub struct BaseFileWriterBuilder<B: FileWriterBuilder> {
```

Comment:
Why not name it `RollingFileWriter`?

Reply:
Because we need to a writer not rolling in position delete writer. This writer always write into single file until flush. And I find that it just a specific form of rolling writer(just ignore the split check), so I combine them.

Reply:
I'm not quite clear. Rolling writer flush by size, while this file writer flushes by caller, I think they are different since the flush method of rolling writer should not be called by caller? 
> Because we need to a writer not rolling in position delete writer.

Why it can't be a rolling writer? The only requirement of pos deletion file is that it should be sorted, and it still can be rolling?

Reply:
> Why it can't be a rolling writer? The only requirement of pos deletion file is that it should be sorted, and it still can be rolling?

I find it will [write into a single file that iceberg java](https://github.com/apache/iceberg/blob/e1fdb5401886f11b6540b917c8ac1d0202015a40/core/src/main/java/org/apache/iceberg/io/SortedPosDeleteWriter.java#L148). But I guess it's optional and will not impact the correctness. 

> I'm not quite clear. Rolling writer flush by size, while this file writer flushes by caller, I think they are different since the flush method of rolling writer should not be called by caller?

Rolling writer can be call flush. flush is manually, if you call flush, the rolling writer will return the result and create a new file writer. The implementation of single file writer is to make the `self.should_split()` always return false. It's also reasonable to split them into different class.🤔

```
if self.should_split() {
            self.close_current_writer().await?;
            self.open_new_writer().await?;
 }
```

Reply:
Got it, thanks!